tests/Integration/ValidationCachingTest.php

```diff
@@ -0,0 +1,250 @@
+
+        Event::fake();
+
+        $this->graphQL(/** @lang GraphQL */ '
+        {
+            foo
+        }
+        ')->assertExactJson([
+            'data' => [
+                'foo' => Foo::THE_ANSWER,
+            ],
+        ]);
+
+        Event::assertDispatchedTimes(CacheMissed::class, 0);
+        Event::assertDispatchedTimes(CacheHit::class, 0);
+        Event::assertDispatchedTimes(KeyWritten::class, 0);
+    }
+
+    public function testConfigMissing(): void
+    {
+        $config = $this->app->make(ConfigRepository::class);
+        $config->set('lighthouse.query_cache.enable', false);
+        $config->set('lighthouse.validation_cache', null);
+
+        Event::fake();
+
+        $this->graphQL(/** @lang GraphQL */ '
+        {
+            foo
+        }
+        ')->assertExactJson([
+            'data' => [
+                'foo' => Foo::THE_ANSWER,
+            ],
+        ]);
+
+        Event::assertDispatchedTimes(CacheMissed::class, 0);
+        Event::assertDispatchedTimes(CacheHit::class, 0);
+        Event::assertDispatchedTimes(KeyWritten::class, 0);
+    }
+
+    public function testErrorsAreNotCached(): void
+    {
+        $config = $this->app->make(ConfigRepository::class);
+        $config->set('lighthouse.query_cache.enable', false);
+        $config->set('lighthouse.validation_cache.enable', true);
+
+        Event::fake();
+
+        $this->graphQL(/** @lang GraphQL */ '
+        {
+            bar
+        }
+        ')->assertGraphQLErrorMessage('Cannot query field "bar" on type "Query".');
+
+        Event::assertDispatchedTimes(CacheMissed::class, 1);
+        Event::assertDispatchedTimes(CacheHit::class, 0);
+        Event::assertDispatchedTimes(KeyWritten::class, 0);
+    }
+
+    public function testDifferentQueriesHasDifferentKeys(): void
+    {
+        $config = $this->app->make(ConfigRepository::class);
+        $config->set('lighthouse.query_cache.enable', false);
+        $config->set('lighthouse.validation_cache.enable', true);
+
+        Event::fake();
+
+        $this->graphQL(/** @lang GraphQL */ '
+        {
+            foo
+        }
+        ')->assertExactJson([
+            'data' => [
+                'foo' => Foo::THE_ANSWER,
+            ],
+        ]);
+
+        $this->graphQL(/** @lang GraphQL */ '
+        {
+            foo
+            foo
+        }
+        ')->assertExactJson([
+            'data' => [
+                'foo' => Foo::THE_ANSWER,
+            ],
+        ]);
+
+        Event::assertDispatchedTimes(CacheMissed::class, 2);
+        Event::assertDispatchedTimes(CacheHit::class, 0);
+        Event::assertDispatchedTimes(KeyWritten::class, 2);
+    }
+
+    public function testSameSchemaAndSameQueryHaveSameKeys(): void
+    {
+        $config = $this->app->make(ConfigRepository::class);
+        $config->set('lighthouse.query_cache.enable', false);
+        $config->set('lighthouse.validation_cache.enable', true);
+
+        $event = Event::fake();
```

Comment:
This is how I would like the other tests to use this.

Reply:
changed `Event::` assertions to `$event->`